geonode/layers/utils.py

```diff
@@ -355,9 +362,17 @@ def get_bbox(filename):
         srs = osr.SpatialReference(wkt=prj_txt)
         srs.AutoIdentifyEPSG()
         epsg_code = srs.GetAuthorityCode(None)
-        datasource = DataSource(filename)
-        layer = datasource[0]
-        bbox_x0, bbox_y0, bbox_x1, bbox_y1 = layer.extent.tuple
+        # can't find epsg code, then check if bbox is within the WGS84 boundary
+        if epsg_code is None and (x_min <= bbox_x0 <= x_max \
```

Comment:
I think this is not valid? What if it is actually a different projection, the bounds would be incorrect in both: value and logic. Also, perhaps it will fail on international dateline.
We have to really track why `srs.GetAuthorityCode(None)` return no EPSG.

@NyakudyaA do you have a suggestion or any alternative to reliably get the epsg_code without using a custom logic like this? Maybe other library?

Reply:
@lucernae the code is used for the following use cases:
- layers with epsg:4326
   - no problem at all.
- layers with epsg non 4326
  - this is where the original DataSource fails to identify the correct epsg code from the shapefile. this PR basically is to resolve this issue.
- layers without epsg, but has a valid projection definition
  - none of the original code and this PR will work. for the short term solution, we just check if the boundary is within the WGS84. if so, then we assume it is a 4326. surely, this solution may not work for all layers without epsg. 

i can only guess that `srs.GetAuthorityCode(None)` fails because some EPSG definition may not present in the gdal's database.